universe/Ship.cpp

```diff
@@ -590,6 +589,7 @@ void Ship::SetShipMetersToMax() {
         if (max_it == m_part_meters.end())
             continue;
 
+        max_it->second.SetCurrent(Meter::LARGE_VALUE);
```

Comment:
Well, in this case apparently this resetting and setting of the max meters by effects does not work as intended. Because without that line you get the problem that space monsters (at least those created by the CreateShip effect) get 0 weapon strength (and in this case it's not a visibility issue, the weapon strength actually *is* 0, tested this by engaging some sentries/maintenance ships etc. in combat).

Furthermore, even empire ships (at least the starting ships you get) also have their weapon strength set to 0. The only reaon why they get their weapon strength correctly set is because of the resupply effect. If I comment out the code section in `Ship::Resupply` that sets the current part meters to max, the starting combat ship has a weapon strength of 0.

Which explains why those space monsters have weapon strength 0 - because, not belonging to any empire, they can't resupply, so their weapon strength stays at the intial 0.

Apparently, if you don't set the max meter to `Meter::LARGE_VALUE` here, the effects which should set the meter don't apply correctly (for whatever reason). With that line things work. I can only guess that something goes awry with the clamping somewhere down the road when the max meter is left at the initial 0 here.

I guessed at this solution because obviously the fuel, shield and structure paired meters also get the max meters set to `Meter::LARGE_VALUE` above, so I thought there might be a reason for this and probably should also be done for the part meters...